bugfixtemplates/check-nullness-of-field/CheckNullnessOfFieldBeforeSetting.java

```diff
@@ -0,0 +1,102 @@
+package bugfixtemplates;
+
+import java.io.File;
+import java.util.Objects;
+
+class CheckNullnessOfFieldBeforeSetting {
+
+    void m1() {
+        if (firstType != null) {
+            firstType = dataType;
+        }
+    }
+
+    void m2() {
+        if (firstType == null) {
+            firstType = dataType;
+        }
+    }
+
+    void m3() {
+        if (Objects.nonNull(firstType)) {
+            firstType = dataType;
+        }
+    }
+
+    void m4() {
+        if (Objects.isNull(firstType)) {
+            firstType = dataType;
+        }
+    }
+
+    void m5() {
+        if (firstType != null) {
+        } else {
+            firstType = dataType;
+        }
+    }
+
+    void m6() {
+        if (firstType == null) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of == on line 40 to != check before setting the field value on line 42 or verify if the check is needed at all.